freeipa/src/test/java/com/sequenceiq/freeipa/flow/freeipa/backup/full/CreateFullBackupHandlerTest.java

```diff
@@ -0,0 +1,138 @@
+        when(stack.getNotDeletedInstanceMetaDataSet()).thenReturn(metaDataSet);
+        Node node1 = mock(Node.class);
+        when(node1.getHostname()).thenReturn("node1");
+        Node node2 = mock(Node.class);
+        when(node2.getHostname()).thenReturn("node2");
+        Set<Node> nodes = Set.of(node1, node2);
+        when(nodeService.mapInstancesToNodes(metaDataSet)).thenReturn(nodes);
+        GatewayConfig gatewayConfig = mock(GatewayConfig.class);
+        when(gatewayConfigService.getPrimaryGatewayConfig(stack)).thenReturn(gatewayConfig);
+
+        StackEvent result = (StackEvent) underTest.doAccept(new HandlerEvent<>(new Event<>(new CreateFullBackupEvent(2L))));
+
+        ArgumentCaptor<OrchestratorStateParams> captor = ArgumentCaptor.forClass(OrchestratorStateParams.class);
+        verify(orchestrator, times(2)).runOrchestratorState(captor.capture());
+        List<OrchestratorStateParams> stateParams = captor.getAllValues();
+        assertThat(stateParams, everyItem(allOf(
+                hasProperty("primaryGatewayConfig", is(gatewayConfig)),
+                hasProperty("state", is("freeipa.backup-full")),
+                hasProperty("allNodes", is(nodes))
+                )));
+        assertThat(stateParams, hasItem(hasProperty("targetHostNames", allOf(
+                hasItem("node1"),
+                iterableWithSize(1)
+        ))));
+        assertThat(stateParams, hasItem(hasProperty("targetHostNames", allOf(
+                hasItem("node2"),
+                iterableWithSize(1)
+        ))));
+
+        assertEquals(2L, result.getResourceId());
+        assertEquals(FullBackupEvent.FULL_BACKUP_SUCCESSFUL_EVENT.event(), result.selector());
+    }
+
+    @Test
+    public void testOrchestratorThrowException() throws CloudbreakOrchestratorFailedException {
+        Stack stack = mock(Stack.class);
+        when(stackService.getByIdWithListsInTransaction(2L)).thenReturn(stack);
+        Set<InstanceMetaData> metaDataSet = Set.of();
+        when(stack.getNotDeletedInstanceMetaDataSet()).thenReturn(metaDataSet);
+        Node node1 = mock(Node.class);
+        when(node1.getHostname()).thenReturn("node1");
+        Set<Node> nodes = Set.of(node1);
+        when(nodeService.mapInstancesToNodes(metaDataSet)).thenReturn(nodes);
+        GatewayConfig gatewayConfig = mock(GatewayConfig.class);
+        when(gatewayConfigService.getPrimaryGatewayConfig(stack)).thenReturn(gatewayConfig);
+        doThrow(new CloudbreakOrchestratorFailedException("tada")).when(orchestrator).runOrchestratorState(any(OrchestratorStateParams.class));
+
+        StackFailureEvent result = (StackFailureEvent) underTest.doAccept(new HandlerEvent<>(new Event<>(new CreateFullBackupEvent(2L))));
+
+        assertEquals(2L, result.getResourceId());
+        assertEquals(FullBackupEvent.FULL_BACKUP_FAILED_EVENT.event(), result.selector());
+        assertEquals("tada", result.getException().getMessage());
+    }
```

Comment:
It might make sense to add a list of verify-s, what gets called and what not.

Reply:
do you have anything exact in your mind? Everything which is mocked must be called or Mockito would fail with lenient issue.
`HostOrchestrator` calls are verified.